From a29870b921f650d67e6bc126200cc675f67c416e Mon Sep 17 00:00:00 2001
From: Zelan Zou <zelan.zou@nxp.com>
Date: Tue, 26 Mar 2024 14:41:53 +0900
Subject: [PATCH 01/11] fix missing meson
Upstream-Status: Inappropriate [workaround, would need a real fix for upstream]
Signed-off-by: Zelan Zou <zelan.zou@nxp.com>
---
 configure | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/configure b/configure
index 82cace1bc9..83bbda986c 100755
--- a/configure
+++ b/configure
@@ -993,10 +993,7 @@ get_pwd() {
     fi
 }

-meson="$(cd pyvenv/bin; get_pwd)/meson"
-if [ -f "$meson$EXESUF" ]; then
-  meson="$meson$EXESUF"
-fi
+meson=`which meson`

 # Conditionally ensure Sphinx is installed.

--
2.34.1
