From ab9bcb6eb22eb01f380c130c6c840d03a07db618 Mon Sep 17 00:00:00 2001
From: Gagandeep Singh <g.singh@nxp.com>
Date: Fri, 25 Jul 2025 15:25:58 +0530
Subject: [PATCH] cmm: support for multilib

Upstream-Status: Pending
Signed-off-by: Gagandeep Singh <g.singh@nxp.com>
---
 configure.ac    | 17 +++++++++++++++++
 src/Makefile.am |  3 ++-
 2 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 252a067..64371a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,6 +20,23 @@ LIBNFCONNTRACK_REQUIRED=0.0.16
 PKG_CHECK_MODULES(LIBNFCONNTRACK, libnetfilter_conntrack >= $LIBNFCONNTRACK_REQUIRED,,
 	AC_MSG_ERROR(Cannot find libnetfilter_conntrack >= $LIBNFNCONNTRACK_REQUIRED))
 
+AC_MSG_CHECKING([for BASELIB environment variable])
+if test -n "$BASELIB"; then
+  # Use the value from the environment if it exists
+  libdir_suffix="$BASELIB"
+else
+  # Default to 'lib' if the variable is not set
+  libdir_suffix="lib"
+fi
+AC_MSG_RESULT([$libdir_suffix])
+
+# Define a new directory variable for Automake
+baselibdir="${libdir}"
+if test "$libdir_suffix" != "lib"; then
+  baselibdir='${exec_prefix}/'"$libdir_suffix"
+fi
+AC_SUBST(baselibdir)
+
 CFLAGS="$CFLAGS $LIBNFCONNTRACK_CFLAGS"
 CMM_LIBS="$LIBNFCONNTRACK_LIBS"
 AC_SUBST(CMM_LIBS)
diff --git a/src/Makefile.am b/src/Makefile.am
index 115b6d3..0e40a52 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -74,12 +74,13 @@ cmm_SOURCES += cpal_fci.c
 endif
 cmm_LDADD = -lpthread -lcli libcmm.la  @CMM_LIBS@ $(CPAL_LIB)
 cmm_DEPENDENCIES = libcmm.la
+cmm_LDFLAGS = -L${top_builddir}
 
 libcmm_sample_SOURCES = libcmm_sample.c
 libcmm_sample_LDADD = libcmm.la
 libcmm_sample_DEPENDENCIES = libcmm.la
 
 
-lib_LTLIBRARIES = libcmm.la
+baselib_LTLIBRARIES = libcmm.la
 libcmm_la_LDFLAGS = -version-info 19:1:0
 libcmm_la_SOURCES = libcmm.c
-- 
2.25.1

