From a61037aa2703d150ed25491ad8bfef06c7d8a709 Mon Sep 17 00:00:00 2001
From: Lauren Post <lauren.post@freescale.com>
Date: Sun, 29 Jun 2014 09:15:32 -0500
Subject: [PATCH] linux-lib:  Disable rng and sahara for imx 6 solox and
 sololite

Signed-off-by: Lauren Post <lauren.post@freescale.com>
---
 rng/Makefile     |    2 +-
 sahara2/Makefile |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/rng/Makefile b/rng/Makefile
index 10b3241..df97bdc 100644
--- a/rng/Makefile
+++ b/rng/Makefile
@@ -1,4 +1,4 @@
-EXCLUDE_LIST:= IMX6Q
+EXCLUDE_LIST:= IMX6Q IMX6S
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 AR = ar -crv
diff --git a/sahara2/Makefile b/sahara2/Makefile
index c7ceeef..1eaa208 100644
--- a/sahara2/Makefile
+++ b/sahara2/Makefile
@@ -1,4 +1,4 @@
-EXCLUDE_LIST:= IMX6Q
+EXCLUDE_LIST:= IMX6Q IMX6S
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 AR = ar -crv
-- 
1.7.9.5

