From 4390e144418a261791bf7420bb7e19c3fe03f488 Mon Sep 17 00:00:00 2001
From: Yong Gan <b45748@freescale.com>
Date: Wed, 27 Aug 2014 18:42:26 +0800
Subject: [PATCH 2/2] disable_libinput_backend.

[DATE]08-27-2014
Signed-off-by Yong Gan <B45748@freescale.com>
---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 05e883d..24f761f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -156,7 +156,7 @@ fi
 
 
 AC_ARG_ENABLE(libinput-backend, [  --disable-libinput-backend],,
-	      enable_libinput_backend=yes)
+	      enable_libinput_backend=no)
 AM_CONDITIONAL([ENABLE_LIBINPUT_BACKEND], [test x$enable_libinput_backend = xyes])
 if test x$enable_libinput_backend = xyes; then
   AC_DEFINE([BUILD_LIBINPUT_BACKEND], [1], [Build the libinput input device backend])
-- 
1.7.9.5

