From a5302ae3377d5b3ebbfca073befb32b78f04661a Mon Sep 17 00:00:00 2001
From: Neena Busireddy <neenareddy.busireddy@nxp.com>
Date: Wed, 19 Dec 2018 14:57:06 -0600
Subject: [PATCH] libteec: Fix GCC 8 format-truncation error

Signed-off-by: Neena Busireddy <neenareddy.busireddy@nxp.com>
---
 libteec/src/teec_trace.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libteec/src/teec_trace.c b/libteec/src/teec_trace.c
index 78b79d6..7901deb 100644
--- a/libteec/src/teec_trace.c
+++ b/libteec/src/teec_trace.c
@@ -73,7 +73,7 @@ int _dprintf(const char *function, int flen, int line, int level,
 	     const char *prefix, const char *fmt, ...)
 {
 	char raw[MAX_PRINT_SIZE];
-	char prefixed[MAX_PRINT_SIZE];
+	char prefixed[MAX_PRINT_SIZE + 10];
 	char *to_print = NULL;
 	const char *func;
 	int err;
@@ -106,7 +106,7 @@ int _dprintf(const char *function, int flen, int line, int level,
 		 */
 		int thread_id = syscall(SYS_gettid);	/* perf issue ? */
 
-		snprintf(prefixed, MAX_PRINT_SIZE,
+		snprintf(prefixed, MAX_PRINT_SIZE + 10,
 			 "%s [%d] %s:%s:%d: %s",
 			 trace_level_strings[level], thread_id, prefix, func,
 			 line, raw);
-- 
2.7.4

