From b657c0e48f565a6a85f2d7f1abe8c2f3474fe8d6 Mon Sep 17 00:00:00 2001
From: Cristinel Panfir <cristinel.panfir@nxp.com>
Date: Mon, 18 Oct 2021 10:52:59 +0000
Subject: [PATCH] CMakeLists.txt: Use CMAKE installation variables

Upstream-Status: Submitted
Signed-off-by: Cristinel Panfir <cristinel.panfir@nxp.com>
---
 src/tim/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tim/CMakeLists.txt b/src/tim/CMakeLists.txt
index cf850cc..dc4f2b0 100644
--- a/src/tim/CMakeLists.txt
+++ b/src/tim/CMakeLists.txt
@@ -75,7 +75,7 @@ target_link_libraries(${TARGET_NAME} PUBLIC
    -Wl,--whole-archive tim_internal -Wl,--no-whole-archive ${EXTERNAL_LIBS})
 
 install(TARGETS ${TARGET_NAME} ${TARGET_NAME}
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
 
 install(DIRECTORY ${CMAKE_SOURCE_DIR}/include/tim/vx 
         DESTINATION ${CMAKE_INSTALL_PREFIX}/include/tim)
@@ -98,7 +98,7 @@ if(TIM_VX_ENABLE_TEST)
         ${CMAKE_CURRENT_SOURCE_DIR}/vx
     )
 
-    install(TARGETS unit_test DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
+    install(TARGETS unit_test DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/)
 endif()
 
 add_subdirectory("utils")
-- 
2.14.2.windows.1

