diff --git a/examples/widgets/graphicsview/CMakeLists.txt b/examples/widgets/graphicsview/CMakeLists.txt
index 86936a8321..0bee531e15 100644
--- a/examples/widgets/graphicsview/CMakeLists.txt
+++ b/examples/widgets/graphicsview/CMakeLists.txt
@@ -1,4 +1,6 @@
-qt_internal_add_example(chip)
+if(QT_FEATURE_opengl)
+    qt_internal_add_example(chip)
+endif()
 qt_internal_add_example(elasticnodes)
 qt_internal_add_example(embeddeddialogs)
 qt_internal_add_example(collidingmice)
diff --git a/examples/widgets/painting/CMakeLists.txt b/examples/widgets/painting/CMakeLists.txt
index 35f3af71b5..839abe541f 100644
--- a/examples/widgets/painting/CMakeLists.txt
+++ b/examples/widgets/painting/CMakeLists.txt
@@ -1,10 +1,12 @@
 qt_internal_add_example(basicdrawing)
 qt_internal_add_example(concentriccircles)
-qt_internal_add_example(affine)
-# qt_internal_add_example(composition) # FIXME: Seems buggy wrt. usesOpenGL function
-qt_internal_add_example(deform)
-qt_internal_add_example(gradients)
-qt_internal_add_example(pathstroke)
+if(QT_FEATURE_opengl)
+    qt_internal_add_example(affine)
+    # qt_internal_add_example(composition)
+    qt_internal_add_example(deform)
+    qt_internal_add_example(gradients)
+    qt_internal_add_example(pathstroke)
+endif()
 qt_internal_add_example(imagecomposition)
 qt_internal_add_example(painterpaths)
 qt_internal_add_example(transformations)
