From 7dde6805a1308b9e03ccdf12d08602787d9f5ee8 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Fri, 26 Jul 2024 12:39:18 -0500
Subject: [PATCH] LF-12809: Use standard linux build assumptions

Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 CMakeLists.txt | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b9996bc..aeca6dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,9 +34,7 @@ add_library (v2xCrypto SHARED
 
 if(UNIFIED_LIBS)
 target_link_libraries(v2xCrypto
-	${HSM_LIBS_DIR}/lib_hsm.so
-	${HSM_LIBS_DIR}/lib_hsm.so.0
-	${HSM_LIBS_DIR}/lib_hsm.so.0.1
+	:lib_hsm.so
 	)
 endif(UNIFIED_LIBS)
 
@@ -52,11 +50,11 @@ target_include_directories(v2xCrypto PRIVATE
 	${HSM_LIBS_DIR}/include/hsm/
 	${LTTNG_DIR}/include/
 	${URCU_DIR}/include/
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
 	)
 
 target_include_directories (v2xCrypto PUBLIC
-	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-	$<INSTALL_INTERFACE:include/v2xsehsm>
+	$<INSTALL_INTERFACE:include>
 	)
 
 set_target_properties (v2xCrypto PROPERTIES
@@ -83,7 +81,7 @@ add_custom_target(doc
 install (TARGETS v2xCrypto
 	EXPORT v2xsehsmTargets
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-	PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/v2xsehsm
+	PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 	)
 
 write_basic_package_version_file (
-- 
2.34.1

