diff -ruN gst-plugins-bad-1.8.1/configure.ac gst-plugins-bad-1.8.1-new/configure.ac
--- gst-plugins-bad-1.8.1/configure.ac	2016-04-20 20:12:05.000000000 +0800
+++ gst-plugins-bad-1.8.1-new/configure.ac	2016-05-11 14:54:36.280070228 +0800
@@ -2833,8 +2833,8 @@
 translit(dnm, m, l) AM_CONDITIONAL(USE_QT, true)
 AG_GST_CHECK_FEATURE(QT, [Qt elements], qt, [
   PKG_CHECK_MODULES(QT, Qt5Core Qt5Gui Qt5Qml Qt5Quick >= 5.4.0, [
-      QT_PATH=`$PKG_CONFIG --variable=exec_prefix Qt5Core`
-      QT_HOST_PATH=`$PKG_CONFIG --variable=host_bins Qt5Core`
+      QT_PATH=$STAGING_DIR`$PKG_CONFIG --variable=exec_prefix Qt5Core`
+      QT_HOST_PATH=$STAGING_DIR`$PKG_CONFIG --variable=host_bins Qt5Core`
       AC_PATH_PROGS(MOC, [moc-qt5 moc], moc, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
       AC_PATH_PROGS(RCC, [rcc-qt5 rcc], rcc, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
       AC_PATH_PROGS(UIC, [uic-qt5 uic], uic, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
