From d78bc1cec51f0d08ac9fec53020b51dfa361d04f Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Thu, 13 May 2021 16:28:14 -0500
Subject: [PATCH] Install header to standard location

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fbbb2fe..3905374 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,7 @@ target_include_directories(v2xCrypto PRIVATE
 
 target_include_directories (v2xCrypto PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-	$<INSTALL_INTERFACE:include/v2xsehsm>
+	$<INSTALL_INTERFACE:include>
 	)
 
 set_target_properties (v2xCrypto PROPERTIES
@@ -69,7 +69,7 @@ add_custom_target(doc
 install (TARGETS v2xCrypto
 	EXPORT v2xsehsmTargets
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-	PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/v2xsehsm
+	PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 	)
 
 write_basic_package_version_file (
-- 
2.17.1

