diff -Nur fsl-gpu-sdk-1.1.0/Samples/GLES1.1/06_Projection/main.h fsl-gpu-sdk-1.1/Samples/GLES1.1/06_Projection/main.h
--- fsl-gpu-sdk-1.1.0/Samples/GLES1.1/06_Projection/main.h	2013-10-09 02:47:03.000000000 +0800
+++ fsl-gpu-sdk-1.1/Samples/GLES1.1/06_Projection/main.h	2014-06-05 22:47:09.517978511 +0800
@@ -30,6 +30,7 @@
 #define __main_h_
 
 #include <GLES/gl.h>
+#include <GLES/glext.h>
 #include <EGL/egl.h>
 
 extern bool drawInOrtho;
diff -Nur fsl-gpu-sdk-1.1.0/Samples/GLES1.1/11_LightingFog/main.h fsl-gpu-sdk-1.1/Samples/GLES1.1/11_LightingFog/main.h
--- fsl-gpu-sdk-1.1.0/Samples/GLES1.1/11_LightingFog/main.h	2013-10-09 02:47:03.000000000 +0800
+++ fsl-gpu-sdk-1.1/Samples/GLES1.1/11_LightingFog/main.h	2014-06-05 22:48:21.033978393 +0800
@@ -30,6 +30,7 @@
 #define __main_h_
 
 #include <GLES/gl.h>
+#include <GLES/glext.h>
 #include <EGL/egl.h>
 
 void PrintHelp();
diff -Nur fsl-gpu-sdk-1.1.0/Samples/GLES1.1/15_ParticleAccelerator/main.h fsl-gpu-sdk-1.1/Samples/GLES1.1/15_ParticleAccelerator/main.h
--- fsl-gpu-sdk-1.1.0/Samples/GLES1.1/15_ParticleAccelerator/main.h	2013-10-09 02:47:03.000000000 +0800
+++ fsl-gpu-sdk-1.1/Samples/GLES1.1/15_ParticleAccelerator/main.h	2014-06-05 22:48:47.505978363 +0800
@@ -30,6 +30,7 @@
 #define __main_h_
 
 #include <GLES/gl.h>
+#include <GLES/glext.h>
 #include <EGL/egl.h>
 
 unsigned getTickCount();
diff -Nur fsl-gpu-sdk-1.1.0/Samples/GLES1.1/16_VertexBufferObjects/main.h fsl-gpu-sdk-1.1/Samples/GLES1.1/16_VertexBufferObjects/main.h
--- fsl-gpu-sdk-1.1.0/Samples/GLES1.1/16_VertexBufferObjects/main.h	2013-10-09 02:47:03.000000000 +0800
+++ fsl-gpu-sdk-1.1/Samples/GLES1.1/16_VertexBufferObjects/main.h	2014-06-05 22:49:13.949976897 +0800
@@ -34,6 +34,7 @@
 #define __main_h_
 
 #include <GLES/gl.h>
+#include <GLES/glext.h>
 #include <EGL/egl.h>
 
 unsigned getTickCount();
