From 6e1a798c89e3b7900c22a6cbab12ba9e870f7fba Mon Sep 17 00:00:00 2001
From: Cristinel Panfir <cristinel.panfir@nxp.com>
Date: Thu, 21 Oct 2021 17:31:11 +0300
Subject: [PATCH] meson.build: Fix subplugin install

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 27a2d9dd..9f6c5aaa 100644
--- a/meson.build
+++ b/meson.build
@@ -95,7 +95,7 @@ nnstreamer_inidir = join_paths(nnstreamer_prefix, get_option('sysconfdir'))
 plugins_install_dir = join_paths(nnstreamer_libdir, 'gstreamer-' + gst_api_verision)
 
 # nnstreamer sub-plugins path
-subplugin_install_prefix = join_paths(nnstreamer_prefix, 'lib', 'nnstreamer')
+subplugin_install_prefix = join_paths(nnstreamer_prefix, get_option('libdir'), 'nnstreamer')
 filter_subplugin_install_dir = join_paths(subplugin_install_prefix, 'filters')
 decoder_subplugin_install_dir = join_paths(subplugin_install_prefix, 'decoders')
 customfilter_install_dir = join_paths(subplugin_install_prefix, 'customfilters')
-- 
2.25.1

