--- chromium-109.0.5414.74_org/content/gpu/BUILD.gn	2023-02-15 14:06:17.765748645 -0700
+++ chromium-109.0.5414.74/content/gpu/BUILD.gn	2023-02-15 14:17:11.000000000 -0700
@@ -8,6 +8,12 @@
 import("//gpu/vulkan/features.gni")
 import("//media/media_options.gni")
 
+declare_args() {
+  # Checks if i.MX GPU is being used
+  use_imxgpu = false
+}
+
+
 # See //content/BUILD.gn for how this works.
 group("gpu") {
   visibility = [ "//content/*" ]  # This is an internal content API.
@@ -138,7 +144,7 @@
 
   # Use DRI on desktop Linux builds.
   if (current_cpu != "s390x" && current_cpu != "ppc64" && is_linux &&
-      !is_castos) {
+      !is_castos && !use_imxgpu) {
     configs += [ "//build/config/linux/dri" ]
   }
 }
