Index: weston-1.11.0/configure.ac
===================================================================
--- weston-1.11.0.orig/configure.ac	2017-06-22 10:52:46.067381326 -0500
+++ weston-1.11.0/configure.ac	2017-06-22 10:52:46.683384381 -0500
@@ -90,12 +90,20 @@
 AM_CONDITIONAL(ENABLE_EGL, test x$enable_egl = xyes)
 if test x$enable_egl = xyes; then
 	AC_DEFINE([ENABLE_EGL], [1], [Build Weston with EGL support])
-	PKG_CHECK_MODULES(EGL, [egl glesv2])
-        PKG_CHECK_MODULES([EGL_TESTS], [egl glesv2 wayland-client wayland-egl])
-	PKG_CHECK_MODULES([GL_RENDERER], [libdrm])
+	PKG_CHECK_MODULES(EGL, [egl])
+	PKG_CHECK_MODULES([EGL_TESTS], [egl wayland-client wayland-egl])
 	COMPOSITOR_MODULES="$COMPOSITOR_MODULES egl"
 fi
 
+AC_ARG_ENABLE(opengl, [  --disable-opengl],,
+			  enable_opengl=$enable_egl)
+AM_CONDITIONAL(ENABLE_OPENGL, test x$enable_egl = xyes -a x$enable_opengl = xyes)
+if test x$enable_egl = xyes -a x$enable_opengl = xyes; then
+	AC_DEFINE([ENABLE_OPENGL], [1], [Build Weston with OpenGL support])
+	PKG_CHECK_MODULES(OPENGL, [glesv2])
+	PKG_CHECK_MODULES([GL_RENDERER], [libdrm])
+fi
+
 AC_ARG_ENABLE(xkbcommon,
 	      AS_HELP_STRING([--disable-xkbcommon], [Disable libxkbcommon
 			      support: This is only useful in environments
@@ -354,9 +362,9 @@
 AC_ARG_ENABLE(simple-egl-clients,
               AS_HELP_STRING([--disable-simple-egl-clients],
                              [do not build the simple EGL clients]),,
-              enable_simple_egl_clients="$enable_egl")
-AM_CONDITIONAL(BUILD_SIMPLE_EGL_CLIENTS, test "x$enable_simple_egl_clients" = "xyes")
-if test x$enable_simple_egl_clients = xyes; then
+              enable_simple_egl_clients="$enable_opengl")
+AM_CONDITIONAL(BUILD_SIMPLE_EGL_CLIENTS, test "x$enable_simple_egl_clients" = "xyes" -a "x$enable_opengl" = "xyes")
+if test x$enable_simple_egl_clients = xyes -a "x$enable_opengl" = "xyes"; then
   PKG_CHECK_MODULES(SIMPLE_EGL_CLIENT,
                     [egl glesv2 wayland-client wayland-egl wayland-cursor])
 fi
@@ -433,13 +441,13 @@
 AM_CONDITIONAL(HAVE_PANGO, test "x$have_pango" = "xyes")
 
 AM_CONDITIONAL(HAVE_CAIRO_GLESV2,
-	       [test "x$have_cairo_egl" = "xyes" -a "x$cairo_modules" = "xcairo-glesv2" -a "x$enable_egl" = "xyes"])
+	       [test "x$have_cairo_egl" = "xyes" -a "x$cairo_modules" = "xcairo-glesv2" -a "x$enable_opengl" = "xyes"])
 
 AM_CONDITIONAL(BUILD_FULL_GL_CLIENTS,
-	       test x$cairo_modules = "xcairo-gl" -a "x$have_cairo_egl" = "xyes" -a "x$enable_egl" = "xyes")
+	       test x$cairo_modules = "xcairo-gl" -a "x$have_cairo_egl" = "xyes" -a "x$enable_opengl" = "xyes")
 
 AM_CONDITIONAL(BUILD_SUBSURFACES_CLIENT,
-	       [test '(' "x$have_cairo_egl" != "xyes" -o "x$cairo_modules" = "xcairo-glesv2" ')' -a "x$enable_simple_egl_clients" = "xyes"])
+	       [test '(' "x$have_cairo_egl" != "xyes" -o "x$cairo_modules" = "xcairo-glesv2" ')' -a "x$enable_simple_egl_clients" = "xyes" -a "x$enable_opengl" = "xyes"])
 
 AM_CONDITIONAL(ENABLE_DESKTOP_SHELL, true)
 
@@ -666,6 +674,7 @@
 
 	Cairo Renderer			${with_cairo}
 	EGL				${enable_egl}
+	OpenGL				${enable_opengl}
 	libxkbcommon			${enable_xkbcommon}
 	xcb_xkb				${have_xcb_xkb}
 	XWayland			${enable_xwayland}
Index: weston-1.11.0/Makefile.am
===================================================================
--- weston-1.11.0.orig/Makefile.am	2017-06-22 10:52:45.703379521 -0500
+++ weston-1.11.0/Makefile.am	2017-06-22 13:29:05.329890578 -0500
@@ -231,10 +231,10 @@
 	ivi-shell/ivi-layout-export.h
 endif
 
-if ENABLE_EGL
+if ENABLE_OPENGL
 module_LTLIBRARIES += gl-renderer.la
 gl_renderer_la_LDFLAGS = -module -avoid-version
-gl_renderer_la_LIBADD = $(COMPOSITOR_LIBS) $(EGL_LIBS)
+gl_renderer_la_LIBADD = $(COMPOSITOR_LIBS) $(EGL_LIBS) $(OPENGL_LIBS)
 gl_renderer_la_CFLAGS =				\
 	$(COMPOSITOR_CFLAGS)			\
 	$(EGL_CFLAGS)				\
@@ -1286,11 +1286,11 @@
 roles_weston_CFLAGS = $(AM_CFLAGS) $(TEST_CLIENT_CFLAGS)
 roles_weston_LDADD = libtest-client.la
 
-if ENABLE_EGL
+if ENABLE_OPENGL
 weston_tests += buffer-count.weston
 buffer_count_weston_SOURCES = tests/buffer-count-test.c
 buffer_count_weston_CFLAGS = $(AM_CFLAGS) $(EGL_TESTS_CFLAGS)
-buffer_count_weston_LDADD = libtest-client.la $(EGL_TESTS_LIBS)
+buffer_count_weston_LDADD = libtest-client.la $(EGL_TESTS_LIBS) $(OPENGL_LIBS)
 endif
 
 if ENABLE_XWAYLAND_TEST
Index: weston-1.11.0/src/main.c
===================================================================
--- weston-1.11.0.orig/src/main.c	2017-06-22 10:52:45.983380910 -0500
+++ weston-1.11.0/src/main.c	2017-06-22 10:52:46.687384401 -0500
@@ -285,11 +285,10 @@
 		"Options for fbdev-backend.so:\n\n"
 		"  --tty=TTY\t\tThe tty to use\n"
 		"  --device=DEVICE\tThe framebuffer device to use\n"
-#if defined(ENABLE_EGL)
+#if defined(ENABLE_OPENGL)
 		"  --use-gl=1\t\tUse the GL renderer (default is 1)\n"
 		"  --use-g2d=1\t\tUse the G2D renderer (default is 0)\n"
 #else
-		"  --use-gl=1\t\tUse the GL renderer (default is 0)\n"
 		"  --use-g2d=1\t\tUse the G2D renderer (default is 1)\n"
 #endif
 		"  --clone-mode\t\tClone display to multiple devices\n\n");
@@ -869,7 +868,7 @@
 {
 	struct weston_fbdev_backend_config config = {
 		.base = {0},
-#ifdef ENABLE_EGL
+#ifdef ENABLE_OPENGL
 		.use_gl = 1,
 		.use_g2d = 0,
 #else
@@ -885,7 +884,9 @@
 	const struct weston_option fbdev_options[] = {
 		{ WESTON_OPTION_INTEGER, "tty", 0, &config.tty },
 		{ WESTON_OPTION_STRING, "device", 0, &config.device },
+#ifdef ENABLE_OPENGL
 		{ WESTON_OPTION_INTEGER, "use-gl", 0, &config.use_gl },
+#endif
 		{ WESTON_OPTION_INTEGER, "use-g2d", 0, &config.use_g2d },
 		{ WESTON_OPTION_BOOLEAN, "clone-mode", 0, &config.clone_mode },
 	};
