Index: xorg-server-1.20.10/hw/xwayland/Makefile.am
===================================================================
--- xorg-server-1.20.10.orig/hw/xwayland/Makefile.am
+++ xorg-server-1.20.10/hw/xwayland/Makefile.am
@@ -24,12 +24,13 @@ Xwayland_SOURCES =				\
 Xwayland_LDADD =				\
 	$(glamor_lib)				\
 	$(XWAYLAND_LIBS)			\
-	$(top_builddir)/glx/libglxvnd.la	\
 	$(XWAYLAND_SYS_LIBS)			\
 	$(top_builddir)/Xext/libXvidmode.la	\
 	$(XSERVER_SYS_LIBS)
 Xwayland_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
 
+Xwayland_DEPENDENCIES =
+
 Xwayland_built_sources =
 
 if GLAMOR_EGL
@@ -56,7 +57,12 @@ Xwayland_built_sources += $(glamor_built
 glamor_lib = $(top_builddir)/glamor/libglamor.la
 
 Xwayland_LDADD += $(GLAMOR_LIBS) $(GBM_LIBS) -lEGL -lGL
-Xwayland_DEPENDENCIES = $(glamor_lib) $(XWAYLAND_LIBS)
+Xwayland_DEPENDENCIES += $(glamor_lib) $(XWAYLAND_LIBS)
+endif
+
+if GLX
+Xwayland_LDADD += $(top_builddir)/glx/libglxvnd.la
+Xwayland_DEPENDENCIES += $(top_builddir)/glx/libglxvnd.la
 endif
 
 Xwayland_built_sources +=					\
