From b3e71db97c981fcb91eb6beaa6a7941fccdc25fc Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Thu, 19 Jan 2017 09:58:51 +0200
Subject: [PATCH 12/31] use dlsym to get function pointers

---
 dispatch/glproc_gl.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/dispatch/glproc_gl.cpp b/dispatch/glproc_gl.cpp
index 8ae40d4..7d162da 100644
--- a/dispatch/glproc_gl.cpp
+++ b/dispatch/glproc_gl.cpp
@@ -221,7 +221,13 @@ _getPublicProcAddress(const char *procName)
 void *
 _getPrivateProcAddress(const char *procName)
 {
-    return (void *)_glXGetProcAddressARB((const GLubyte *)procName);
+    void *proc;
+    proc = _getPublicProcAddress(procName);
+    if (!proc) {
+        proc = (void *)_glXGetProcAddressARB((const GLubyte *)procName);
+    }
+
+    return proc;
 }
 
 
-- 
2.7.4

