From 172058a38ebe53080264ceebd4c10f18ea0b14fd Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Thu, 19 Jan 2017 11:49:17 +0200
Subject: [PATCH 14/31] MGS-469: GPU hangs when retracing OGLESParticles on
 mx6sx

---
 retrace/retrace.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/retrace/retrace.py b/retrace/retrace.py
index d506abc..164944e 100644
--- a/retrace/retrace.py
+++ b/retrace/retrace.py
@@ -465,7 +465,9 @@ class Retracer:
         print '    if (retrace::verbosity >= 0) {'
         print '        retrace::unsupported(call);'
         print '    }'
-        print '    return;'
+        # MGS-469: GPU hangs when retracing OGLESParticles on mx6sx
+        # Workaround: do not `return`
+        #print '    return;'
 
     def extractArg(self, function, arg, arg_type, lvalue, rvalue):
         ValueAllocator().visit(arg_type, lvalue, rvalue)
-- 
2.7.4

