From 2646cdaf6ae97d205251634196d23679e9b026bb Mon Sep 17 00:00:00 2001
From: Adrian Negreanu <groleo@gmail.com>
Date: Wed, 29 Mar 2017 14:39:17 +0300
Subject: [PATCH 16/31] don't recreate the EGL surface on resize

---
 retrace/glws_egl_xlib.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/retrace/glws_egl_xlib.cpp b/retrace/glws_egl_xlib.cpp
index f4c363c..e50e6dd 100644
--- a/retrace/glws_egl_xlib.cpp
+++ b/retrace/glws_egl_xlib.cpp
@@ -173,6 +173,7 @@ public:
 
         eglWaitNative(EGL_CORE_NATIVE_ENGINE);
 
+#if 0
         /*
          * Some implementations won't update the backbuffer unless we recreate
          * the EGL surface.
@@ -193,6 +194,7 @@ public:
 
         assert(eglWidth == width);
         assert(eglHeight == height);
+#endif
     }
 
     void show(void) {
-- 
2.7.4

