--- a/CMakeLists.txt	2017-04-26 12:05:19.511369999 +0300
+++ b/CMakeLists.txt	2017-04-26 12:06:46.651930949 +0300
@@ -144,7 +144,14 @@
 elseif (APPLE)
     set (ENABLE_EGL false)
 else ()
-    find_package (X11)
+
+    if (NOT DISABLE_X11)
+        message ("DISABLE_X11 is not set")
+        find_package (X11)
+    else ()
+        message ("DISABLE_X11 is set")
+        set (X11_FOUND false)
+    endif ()
 
     if (X11_FOUND)
         include_directories (${X11_INCLUDE_DIR})
