Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt	2017-06-13 16:30:37.707992787 -0500
+++ git/CMakeLists.txt	2017-06-13 16:40:37.534967169 -0500
@@ -40,6 +40,8 @@
 
 option (ENABLE_EGL "Enable EGL support." ON)
 
+option (ENABLE_MULTIARCH "Enable multiarch support." ON)
+
 option (ENABLE_WAFFLE "Enable WAFFLE support." OFF)
 
 option (ENABLE_FRAME_POINTER "Disable frame pointer omission" ON)
@@ -377,7 +379,7 @@
 ##############################################################################
 # Installation directories
 
-if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if (ENABLE_MULTIARCH AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
     # Debian multiarch support
     execute_process(COMMAND dpkg-architecture -qDEB_HOST_MULTIARCH
         OUTPUT_VARIABLE ARCH_SUBDIR
