From 0c0de822331401f6463e47088f0750505378a4a0 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Thu, 13 May 2021 16:20:55 -0500
Subject: [PATCH] Ignore v2xsehsmConfig.cmake

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cde835f..a636ffa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,8 +5,6 @@ project (vtest)
 
 set (vtest_VERSION 0.5)
 
-find_package(v2xsehsm CONFIG REQUIRED)
-
 configure_file (
 	${PROJECT_SOURCE_DIR}/version.h.in
 	${CMAKE_CURRENT_SOURCE_DIR}/include/version.h
-- 
2.17.1

