From ad26420455f9af96e1f660b54f5b37782aefe22a Mon Sep 17 00:00:00 2001
From: Yong Gan <yong.gan@nxp.com>
Date: Tue, 19 Jul 2016 10:03:51 +0800
Subject: [PATCH] drm-update-arm

---
 xf86drm.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/xf86drm.h b/xf86drm.h
index 481d882..4d8aae1 100644
--- a/xf86drm.h
+++ b/xf86drm.h
@@ -473,7 +473,12 @@ do {	register unsigned int __old __asm("o0");		\
 #endif /* __GNUC__ >= 2 */
 
 #ifndef DRM_CAS
-#define DRM_CAS(lock,old,new,ret) do { ret=1; } while (0) /* FAST LOCK FAILS */
+#define DRM_CAS(lock, old, new, __ret)                             \
+    do {                                                           \
+           __ret = !__sync_bool_compare_and_swap(                  \
+                           &__drm_dummy_lock(lock),                \
+                           old, new);                              \
+    } while (0)
 #endif
 
 #if defined(__alpha__)
-- 
1.9.1

