Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt	2017-02-23 14:18:38.630086751 -0600
+++ git/CMakeLists.txt	2017-02-23 14:48:27.000000000 -0600
@@ -271,7 +271,10 @@
 endif()
 
 if(UNIX)
-    install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/vulkan" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    option(INSTALL_VULKAN_HEADERS "Install Vulkan headers" ON)
+    if (INSTALL_VULKAN_HEADERS)
+        install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/vulkan" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    endif()
 endif()
 
 # loader: Generic VULKAN ICD loader
