diff -Naur qtbase-opensource-src-5.2.1_org/config.tests/qpa/directfb/directfb.cpp qtbase-opensource-src-5.2.1/config.tests/qpa/directfb/directfb.cpp
--- qtbase-opensource-src-5.2.1_org/config.tests/qpa/directfb/directfb.cpp      2014-04-10 11:11:02.004158410 -0500
+++ qtbase-opensource-src-5.2.1/config.tests/qpa/directfb/directfb.cpp  2014-04-10 11:12:28.917535069 -0500
@@ -40,11 +40,11 @@
 ****************************************************************************/

 #include <directfb.h>
-
+/*
 #ifdef __typeof__
 #error DirectFB headers are unclean and cannot compile
 #endif
-
+*/
 int main(int, char **)
 {
     return 0;