diff --git a/kernel-module-imx-gpu-viv-5.0.11.p8.7/Makefile b/kernel-module-imx-gpu-viv-6.2.1.p0/Makefile
index 30c799b..d0ea28b 100644
--- a/Makefile
+++ b/Makefile
@@ -4,6 +4,8 @@ SRC := $(shell pwd)
 
 all:
 	$(MAKE) -C $(KERNEL_SRC) M=$(SRC)/kernel-module-imx-gpu-viv-src AQROOT=${PWD}/kernel-module-imx-gpu-viv-src
+	cp $(SRC)/kernel-module-imx-gpu-viv-src/Module.symvers $(PWD)
+	cp $(SRC)/kernel-module-imx-gpu-viv-src/modules.order $(PWD)
 
 modules_install:
 	$(MAKE) -C $(KERNEL_SRC) M=$(SRC)/kernel-module-imx-gpu-viv-src modules_install
