From 164697b4a666f181ea2c9e2c69e66139b2c7cb62 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Thu, 30 Nov 2023 15:01:09 -0600
Subject: [PATCH] Fix build for non-existent BuildTfLiteIntArray

Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 simple_delegate.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/simple_delegate.cc b/simple_delegate.cc
index 664028d..51bd5b7 100644
--- a/simple_delegate.cc
+++ b/simple_delegate.cc
@@ -110,7 +110,7 @@ TfLiteStatus DelegatePrepare(TfLiteContext* context,
 
   return context->ReplaceNodeSubsetsWithDelegateKernels(
       context, delegate_kernel_registration,
-      BuildTfLiteIntArray(supported_nodes).get(), base_delegate);
+      BuildTfLiteArray(supported_nodes).get(), base_delegate);
 }
 }  // namespace
 
-- 
2.34.1

