Index: weston-1.11.0/src/g2d-renderer.c
===================================================================
--- weston-1.11.0.orig/src/g2d-renderer.c	2017-04-07 16:34:42.083154031 -0500
+++ weston-1.11.0/src/g2d-renderer.c	2017-04-07 16:53:36.476221370 -0500
@@ -38,7 +38,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <g2dExt.h>
-#include <HAL/gc_hal_eglplatform.h>
+#include <HAL/gc_hal_base.h>
 
 #include "compositor.h"
 #include "g2d-renderer.h"
