From 1afa3fd6fa16ce261566de9d319a1945a34188fd Mon Sep 17 00:00:00 2001
From: Neena Busireddy <neenareddy.busireddy@nxp.com>
Date: Wed, 21 Jan 2026 11:01:40 -0600
Subject: [PATCH] fixup! Fix source sha256 mismatch issue for KleidiAI

Signed-off-by: Neena Busireddy <neenareddy.busireddy@nxp.com>
---
Upstream-Status: Inappropriate

 tflite/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tflite/CMakeLists.txt b/tflite/CMakeLists.txt
index bf6bdaf..2352ec4 100644
--- a/tflite/CMakeLists.txt
+++ b/tflite/CMakeLists.txt
@@ -79,7 +79,7 @@ option(TFLITE_ENABLE_EXTERNAL_DELEGATE "Enable External Delegate backend" ON)
 
 option(TFLITE_KERNEL_TEST "Enable tflite kernel unit test" OFF)
 
-if(NOT ${CMAKE_CROSSCOMPILING})
+if(${CMAKE_CROSSCOMPILING})
   set(TFLITE_HOST_TOOLS_DIR "" CACHE PATH "Host tools directory")
   if(TFLITE_HOST_TOOLS_DIR STREQUAL "")
     message(FATAL_ERROR "When cross-compiling, some tools need to be available to run on the host (current required tools: flatc). Please specify where those binaries can be found by using -DTFLITE_HOST_TOOLS_DIR=<flatc_dir_path>.")
-- 
2.34.1

