From 41329721d6a1b82de88b8ccf03d4f2339f771910 Mon Sep 17 00:00:00 2001
From: Elliot Chen <elliot.chen@nxp.com>
Date: Fri, 27 Feb 2026 19:45:22 +0900
Subject: [PATCH] pipewiresrc: update per-plane stride and offset according to
 chunk info

---
 src/gst/gstpipewiresrc.c | 38 ++++++++++++++++++++++++++++++--------
 1 file changed, 30 insertions(+), 8 deletions(-)

Upstream-Status: Pending

diff --git a/src/gst/gstpipewiresrc.c b/src/gst/gstpipewiresrc.c
index b0de17dfd..d9e770b14 100644
--- a/src/gst/gstpipewiresrc.c
+++ b/src/gst/gstpipewiresrc.c
@@ -829,14 +829,36 @@ static GstBuffer *dequeue_buffer(GstPipeWireSrc *pwsrc)
                              info->offset,
                              info->stride);
 
-    for (i = 0; i < MIN (n_datas, n_planes); i++) {
-      struct spa_data *d = &b->buffer->datas[i];
-      /* don't add the chunk offset here, this is done below when we
-       * share/copy the memory in the target buffer below */
-      meta->offset[i] = video_size;
-      meta->stride[i] = d->chunk->stride;
-
-      video_size += d->chunk->size;
+    if (n_datas < n_planes) {
+      /* If the buffer number is smaller than the plane number,
+      * we should update the stride and offset for all planes.
+      */
+      gsize offset = 0;
+      guint stride;
+      for (i = 0; i < n_planes; i++) {
+        if (i < b->buffer->n_datas) {
+          struct spa_data *d = &b->buffer->datas[i];
+          stride = d->chunk->stride;
+        } else {
+          stride = gst_video_format_info_extrapolate_stride (info->finfo,
+              i, b->buffer->datas[0].chunk->stride);
+        }
+
+        meta->stride[i] = stride;
+        meta->offset[i] = offset;
+        offset += stride * GST_VIDEO_FORMAT_INFO_SCALE_HEIGHT (info->finfo,
+            i, GST_VIDEO_INFO_HEIGHT(info));
+      }
+    } else {
+      for (i = 0; i < MIN (n_datas, n_planes); i++) {
+        struct spa_data *d = &b->buffer->datas[i];
+        /* don't add the chunk offset here, this is done below when we
+        * share/copy the memory in the target buffer below */
+        meta->offset[i] = video_size;
+        meta->stride[i] = d->chunk->stride;
+
+        video_size += d->chunk->size;
+      }
     }
   }
 
-- 
2.34.1

