From a6b45ba9e59bd188004855cdf6c8b9a02a0c3618 Mon Sep 17 00:00:00 2001
From: Hou Qi <qi.hou@nxp.com>
Date: Fri, 13 Sep 2024 22:48:10 +0900
Subject: [PATCH 06/21] V4L2VideoDecoder: Use correct plane size and bytesused

Upstream-Status: Inappropriate [NXP specific]
---
 media/gpu/v4l2/v4l2_queue.cc | 20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/media/gpu/v4l2/v4l2_queue.cc b/media/gpu/v4l2/v4l2_queue.cc
index 9168606742d7f..400f5f12b8676 100644
--- a/media/gpu/v4l2/v4l2_queue.cc
+++ b/media/gpu/v4l2/v4l2_queue.cc
@@ -858,7 +858,10 @@ size_t V4L2WritableBufferRef::GetPlaneSize(const size_t plane) const {
     return 0;
   }
 
-  return buffer_data_->v4l2_buffer_.m.planes[plane].length;
+  if (V4L2_TYPE_IS_MULTIPLANAR(buffer_data_->v4l2_buffer_.type))
+    return buffer_data_->v4l2_buffer_.m.planes[plane].length;
+  else
+    return buffer_data_->v4l2_buffer_.length;
 }
 
 void V4L2WritableBufferRef::SetPlaneSize(const size_t plane,
@@ -918,7 +921,10 @@ void V4L2WritableBufferRef::SetPlaneBytesUsed(const size_t plane,
     return;
   }
 
-  buffer_data_->v4l2_buffer_.m.planes[plane].bytesused = bytes_used;
+  if (V4L2_TYPE_IS_MULTIPLANAR(buffer_data_->v4l2_buffer_.type))
+    buffer_data_->v4l2_buffer_.m.planes[plane].bytesused = bytes_used;
+  else
+    buffer_data_->v4l2_buffer_.bytesused = bytes_used;
 }
 
 size_t V4L2WritableBufferRef::GetPlaneBytesUsed(const size_t plane) const {
@@ -930,7 +936,10 @@ size_t V4L2WritableBufferRef::GetPlaneBytesUsed(const size_t plane) const {
     return 0;
   }
 
-  return buffer_data_->v4l2_buffer_.m.planes[plane].bytesused;
+  if (V4L2_TYPE_IS_MULTIPLANAR(buffer_data_->v4l2_buffer_.type))
+    return buffer_data_->v4l2_buffer_.m.planes[plane].bytesused;
+  else
+    return buffer_data_->v4l2_buffer_.bytesused;
 }
 
 void V4L2WritableBufferRef::SetPlaneDataOffset(const size_t plane,
@@ -1037,7 +1046,10 @@ size_t V4L2ReadableBuffer::GetPlaneBytesUsed(const size_t plane) const {
     return 0;
   }
 
-  return buffer_data_->v4l2_planes_[plane].bytesused;
+  if (V4L2_TYPE_IS_MULTIPLANAR(buffer_data_->v4l2_buffer_.type))
+    return buffer_data_->v4l2_planes_[plane].bytesused;
+  else
+    return buffer_data_->v4l2_buffer_.bytesused;
 }
 
 size_t V4L2ReadableBuffer::GetPlaneDataOffset(const size_t plane) const {
-- 
2.34.1

