From eb751ca901de9a5c1726afbd7da670c2aec25fce Mon Sep 17 00:00:00 2001
From: Hou Qi <qi.hou@nxp.com>
Date: Wed, 15 Oct 2025 18:36:11 +0900
Subject: [PATCH 12/21] V4L2VideoDecoder: Use dlopen to dynamically use g2d api

Upstream-Status: Inappropriate [NXP specific]
---
 media/gpu/v4l2/v4l2_queue.cc | 49 ++++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/media/gpu/v4l2/v4l2_queue.cc b/media/gpu/v4l2/v4l2_queue.cc
index e4d2e4c0830c1..47006144aece2 100644
--- a/media/gpu/v4l2/v4l2_queue.cc
+++ b/media/gpu/v4l2/v4l2_queue.cc
@@ -30,8 +30,11 @@
 #include "third_party/perfetto/include/perfetto/tracing/track.h"
 
 #if BUILDFLAG(USE_IMXG2D)
+#include "g2d.h"
+#include "g2dExt.h"
 #include <linux/dma-buf.h>
 #include <time.h>
+#include <dlfcn.h>
 #endif
 
 namespace media {
@@ -166,6 +169,27 @@ std::vector<base::ScopedFD> GetDmabufsForV4L2Buffer(
 
 }  // namespace
 
+#if BUILDFLAG(USE_IMXG2D)
+void *dl_handle = NULL;
+
+typedef int (*g2d_api_open)(void **handle);
+typedef int (*g2d_api_close)(void *handle);
+typedef int (*g2d_api_free)(struct g2d_buf *buf);
+typedef struct g2d_buf* (*g2d_api_alloc)(int size, int cacheable);
+typedef int (*g2d_api_buf_export_fd)(struct g2d_buf *);
+typedef int (*g2d_api_blitEx)(void *handle, struct g2d_surfaceEx *srcEx, struct g2d_surfaceEx *dstEx);
+
+#define G2D_LIB_NAME "/usr/lib/libg2d.so.2"
+#define G2D_API_SYM(name) g2d_api_##name g2d_##name = nullptr
+G2D_API_SYM(open);
+G2D_API_SYM(close);
+G2D_API_SYM(free);
+G2D_API_SYM(alloc);
+G2D_API_SYM(buf_export_fd);
+G2D_API_SYM(blitEx);
+#undef G2D_API_SYM
+#endif
+
 V4L2ExtCtrl::V4L2ExtCtrl(uint32_t id) {
   memset(&ctrl, 0, sizeof(ctrl));
   ctrl.id = id;
@@ -1269,6 +1293,31 @@ V4L2Queue::V4L2Queue(base::PassKey<PassKey>,
       << " support requests.";
 
 #if BUILDFLAG(USE_IMXG2D)
+  if (dl_handle == NULL) {
+    dl_handle = dlopen(G2D_LIB_NAME,
+                        RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE | RTLD_LAZY);
+    if (!dl_handle) {
+      VLOGF(1) << "Failed to dlopen " << G2D_LIB_NAME;
+      return;
+    }
+
+#define G2D_API_DLSYM(lib, name)                                               \
+    do {                                                                       \
+      g2d_##name = reinterpret_cast<g2d_api_##name>(dlsym(lib, "g2d_" #name)); \
+      if (!(g2d_##name))                                                       \
+        VLOGF(1) << "Failed to dlsym g2d_" #name;                              \
+    } while (0)
+
+    G2D_API_DLSYM(dl_handle, open);
+    G2D_API_DLSYM(dl_handle, close);
+    G2D_API_DLSYM(dl_handle, free);
+    G2D_API_DLSYM(dl_handle, alloc);
+    G2D_API_DLSYM(dl_handle, buf_export_fd);
+    G2D_API_DLSYM(dl_handle, blitEx);
+
+    dlclose(dl_handle);
+  }
+
   g2d_handle = NULL;
   if(g2d_open && g2d_open(&g2d_handle))
     VLOGF(1) << "g2d_open fail";
-- 
2.34.1

