diff -Naur chromium-52.0.2743.76_org/services/shell/runner/host/linux_sandbox.cc chromium-52.0.2743.76/services/shell/runner/host/linux_sandbox.cc
--- chromium-52.0.2743.76_org/services/shell/runner/host/linux_sandbox.cc	2016-10-05 09:19:52.025886036 -0500
+++ chromium-52.0.2743.76/services/shell/runner/host/linux_sandbox.cc	2016-10-05 09:33:59.000000000 -0500
@@ -38,6 +38,7 @@
   const sandbox::syscall_broker::BrokerProcess* broker_process =
       static_cast<const sandbox::syscall_broker::BrokerProcess*>(aux);
   switch (args.nr) {
+#if !defined(__aarch64__)
     case __NR_access:
       return broker_process->Access(reinterpret_cast<const char*>(args.args[0]),
                                     static_cast<int>(args.args[1]));
@@ -63,6 +64,7 @@
     default:
       RAW_CHECK(false);
       return -ENOSYS;
+#endif
   }
 }
 
