From 801aa0aad1f62fb41d8f28d3f2e0bf836d9aeec7 Mon Sep 17 00:00:00 2001
From: Ahsan Hussain <ahsan_hussain@mentor.com>
Date: Mon, 18 Sep 2017 14:16:13 +0500
Subject: [PATCH 1/1] CMakeLists: add include paths for xcb and Xlib

---
 vktrace/vktrace_viewer/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/vktrace/vktrace_viewer/CMakeLists.txt b/vktrace/vktrace_viewer/CMakeLists.txt
index 7127c7f..3a203ad 100644
--- a/vktrace/vktrace_viewer/CMakeLists.txt
+++ b/vktrace/vktrace_viewer/CMakeLists.txt
@@ -41,6 +41,9 @@ endif()
 
 find_package(Qt5 COMPONENTS Widgets Gui Core Svg QUIET)
 
+find_path(XCB_INCLUDE_DIR xcb/xcb.h DOC "Path to xcb/xcb.h")
+find_path(X11_XLIB_INCLUDE_DIR X11/Xlib.h DOC "Path to X11/Xlib.h")
+
 if(NOT Qt5_FOUND)
 # After Qt5.6 is installed, you may need to add the following to the cmake command line:
 # -DCMAKE_PREFIX_PATH=C:\\Qt\\5.6\\msvc2015_64\\
@@ -60,6 +63,8 @@ include_directories(
     ${CMAKE_CURRENT_BINARY_DIR}
     ${CMAKE_BINARY_DIR}
     ${Qt5Widgets_INCLUDE_DIRS}
+    ${X11_XLIB_INCLUDE_DIR}
+    ${XCB_INCLUDE_DIR}
 )
 
 set(SRC_LIST
-- 
2.11.1

