diff --git a/tests/egl/egl-nok-swap-region.c b/tests/egl/egl-nok-swap-region.c
index 4d2734d..fb4ad5f 100644
--- a/tests/egl/egl-nok-swap-region.c
+++ b/tests/egl/egl-nok-swap-region.c
@@ -66,10 +66,10 @@ draw(struct egl_state *state)
 		{ 55, 55, red },
 		{ 55, state->height - 55, green },
 	};
-	PFNEGLSWAPBUFFERSREGIONNOK swap_buffers_region;
+	PFNEGLSWAPBUFFERSREGIONNOKPROC swap_buffers_region;
 	int i;
 
-	swap_buffers_region = (PFNEGLSWAPBUFFERSREGIONNOK) 
+	swap_buffers_region = (PFNEGLSWAPBUFFERSREGIONNOKPROC) 
 		eglGetProcAddress("eglSwapBuffersRegionNOK");
 
 	if (swap_buffers_region == NULL) {
