--- gst-plugins-bad-1.10.0/configure.ac	2016-11-24 15:40:16.192709205 +0800
+++ gst-plugins-bad-1.10.0-new/configure.ac	2016-11-24 15:44:55.735811203 +0800
@@ -2300,7 +2300,7 @@
     PKG_CHECK_MODULES(WAYLAND_PROTOCOLS, wayland-protocols >= 1.4, [
       if test "x$wayland_scanner" != "x"; then
         HAVE_WAYLAND="yes"
-        AC_SUBST(WAYLAND_PROTOCOLS_DATADIR, `$PKG_CONFIG --variable=pkgdatadir wayland-protocols`)
+        AC_SUBST(WAYLAND_PROTOCOLS_DATADIR, $PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=pkgdatadir wayland-protocols`)
       else
         AC_MSG_RESULT([wayland-scanner is required to build the wayland plugin])
         HAVE_WAYLAND="no"
