Upstream-Status: Inappropriate [workaround]

Index: 3rdparty/chromium/build/config/compiler/BUILD.gn
===================================================================
--- 3rdparty.orig/chromium/build/config/compiler/BUILD.gn
+++ 3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -449,10 +449,12 @@ config("compiler") {
         "-Wno-psabi", # skia
         "-Wno-multichar", # crashpad
         "-Wno-format-zero-length", # user_agent.cc
+        "-Wno-error=implicit-function-declaration",
       ]
       cflags_cc += [
         "-Wno-deprecated-copy",
         "-Wno-expansion-to-defined",
+        "-Wno-error=implicit-function-declaration",
       ]
     }
   }
