Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt	2017-02-23 14:06:56.000000000 -0600
+++ git/CMakeLists.txt	2017-02-23 14:07:06.000000000 -0600
@@ -272,7 +272,10 @@
 endif()
 
 if(UNIX)
-    install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/vulkan" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    option(INSTALL_VULKAN_HEADERS "Install Vulkan headers" ON)
+    if (INSTALL_VULKAN_HEADERS)
+        install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/vulkan" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    endif()
 endif()
 
 # loader: Generic VULKAN ICD loader
