diff -Nur a/utils/Makefile b/utils/Makefile
--- a/utils/Makefile	2014-03-19 10:18:54.086971402 +0800
+++ b/utils/Makefile	2014-03-19 10:19:37.014970217 +0800
@@ -22,24 +22,24 @@ include $(top_srcdir)/include/mk/env_pre.mk
 
-INSTALL_TARGETS         += ffsb
+#INSTALL_TARGETS         += ffsb
 
-FFSBDIR			:= ffsb-6.0-rc2
-FILTER_OUT_DIRS		:= $(FFSBDIR)
-FFSB			:= $(FFSBDIR)/ffsb
+#FFSBDIR			:= ffsb-6.0-rc2
+#FILTER_OUT_DIRS		:= $(FFSBDIR)
+#FFSB			:= $(FFSBDIR)/ffsb
 
-$(FFSB): $(FFSBDIR)
-	cd $^; ./configure
-	$(MAKE) -C $^ -f "$(abs_srcdir)/$^/Makefile" all
-	cp $(FFSBDIR)/ffsb ffsb
+#$(FFSB): $(FFSBDIR)
+#	cd $^; ./configure
+#	$(MAKE) -C $^ -f "$(abs_srcdir)/$^/Makefile" all
+#	cp $(FFSBDIR)/ffsb ffsb
 
-trunk-all: $(FFSB)
+#trunk-all: $(FFSB)
 
-trunk-clean:: | ffsb-clean
+#trunk-clean:: | ffsb-clean
 
-ffsb-clean:: $(FFSBDIR)
-	$(MAKE) -C $^ -f "$(abs_srcdir)/$^/Makefile" clean
-	rm -rf ffsb
+#ffsb-clean:: $(FFSBDIR)
+#	$(MAKE) -C $^ -f "$(abs_srcdir)/$^/Makefile" clean
+#	rm -rf ffsb
 
 
 include $(top_srcdir)/include/mk/generic_trunk_target.mk
