Index: chromium-80.0.3987.132/ui/ozone/platform/wayland/BUILD.gn
===================================================================
--- chromium-80.0.3987.132.orig/ui/ozone/platform/wayland/BUILD.gn
+++ chromium-80.0.3987.132/ui/ozone/platform/wayland/BUILD.gn
@@ -260,7 +260,9 @@ source_set("test_support") {
 source_set("wayland_unittests") {
   testonly = true
 
-  assert(use_wayland_gbm)
+  if (use_wayland_gbm) {
+    assert(use_wayland_gbm)
+  }
 
   sources = [
     "gpu/wayland_surface_factory_unittest.cc",
@@ -295,9 +297,12 @@ source_set("wayland_unittests") {
     "//ui/ozone:platform",
     "//ui/ozone:test_support",
     "//ui/ozone/common/linux:drm",
-    "//ui/ozone/common/linux:gbm",
     "//ui/platform_window/platform_window_handler",
   ]
+
+    if (use_wayland_gbm) {
+     deps += [ "//ui/ozone/common/linux:gbm" ]
+   }
 
   import("//ui/base/ui_features.gni")
   if (use_xkbcommon) {
@@ -306,8 +311,11 @@ source_set("wayland_unittests") {
 
   defines = [
     "WL_HIDE_DEPRECATED",
-    "WAYLAND_GBM",
   ]
+
+  if (use_wayland_gbm) {
+    defines += [ "WAYLAND_GBM" ]
+  }
 }
 
 fuzzer_test("wayland_buffer_fuzzer") {
